src/meta/src/stream/stream_graph/fragment.rs

```diff
@@ -137,6 +137,8 @@ impl BuildingFragment {
             NodeBody::Dml(dml_node) => {
                 dml_node.table_id = table_id;
                 dml_node.table_version_id = job.table_version_id().unwrap();
+                let table = job.table_mut().unwrap();
+                table.dml_fragment_id = Some(fragment_id);
```

Comment:
I'm wondering if we can apply a similar approach to how we fill the `fragment_id`, so that the `StreamingJob` in `Graph::new` can still be immutable, which improves the readability and prevent it from further being misused.

https://github.com/risingwavelabs/risingwave/blob/77c60b1682822f612c367fc2e58546e475d241fa/src/meta/src/rpc/ddl_controller.rs#L347